lib/veil/credential_collection/base.rb

```diff
@@ -146,6 +150,21 @@ def add(*args)
       end
       alias_method :<<, :add
 
+      # Add the contents of a file as a credential after
+      # verifying that the file can be read.
+      # Usage:
+      #  add_from_file(filename, "secretname")
+      #  add_from_file(filename, "groupname", "secretname")
+      #
+      #  `add`'s options are not supported
+      #
+      def add_from_file(filepath, *args)
```

Comment:
Good point, in this context we will not likely be using this for picking up anything other than keys. I'll just make it freeze - if we find cases where that isn't what we want, we can revisit or make the interface more explicit.